ApplicationLibCode/ProjectDataModel/EnsembleFileSet/RimEnsembleFileSet.cpp

```diff
@@ -0,0 +1,366 @@
+/////////////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (C) 2025 Equinor ASA
+//
+//  ResInsight is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  ResInsight is distributed in the hope that it will be useful, but WITHOUT ANY
+//  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+//  FITNESS FOR A PARTICULAR PURPOSE.
+//
+//  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+//  for more details.
+//
+/////////////////////////////////////////////////////////////////////////////////
+
+#include "RimEnsembleFileSet.h"
+
+#include "Ensemble/RiaEnsembleImportTools.h"
+#include "RiaEnsembleNameTools.h"
+#include "RiaFilePathTools.h"
+#include "RiaStdStringTools.h"
+#include "RiaTextStringTools.h"
+
+#include "RimEnsembleFileSetCollection.h"
+#include "RimProject.h"
+
+#include "cafCmdFeatureMenuBuilder.h"
+#include "cafPdmFieldScriptingCapability.h"
+#include "cafPdmUiComboBoxEditor.h"
+#include "cafPdmUiLineEditor.h"
+
+CAF_PDM_SOURCE_INIT( RimEnsembleFileSet, "EnsembleFileSet" );
+
+namespace internal
+{
+QString placeholderString()
+{
+    return "*";
+}
+} // namespace internal
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RimEnsembleFileSet::RimEnsembleFileSet()
+    : fileSetChanged( this )
+    , nameChanged( this )
+    , m_useKey1( false )
+    , m_useKey2( false )
+
+{
+    CAF_PDM_InitObject( "Ensemble", ":/CreateGridCaseGroup16x16.png", "", "" );
+
+    CAF_PDM_InitField( &m_pathPattern, "PathPattern", QString(), "Path Pattern", "", "", "" );
+    CAF_PDM_InitField( &m_realizationSubSet, "RealizationSubSet", QString(), "Realization SubSet", "", "", "" );
```

Comment:
Avoid CamelCase in user-visible strings. Maybe "Realization Filter" is better?

Reply:
Fixed